File: 프로그래머스/2/1844. 게임 맵 최단거리/게임 맵 최단거리.js
function solution(maps) {
    const answer = -1;
    const dx = [1,-1,0,0]
    const dy = [0,0,1,-1]
    const n = maps.length
    const m = maps[0].length
    let visited = Array(n).fill().map((_)=>Array(m).fill(false))
    let q = [[0,0,1]]
    visited[0][0] = true
    while(q.length > 0){
        const [x,y,count] = q.shift()
        if (x === m-1 && y === n-1){
            return count
        }
        
        for (let i = 0; i<4; i++){
            const nx = x + dx[i]
            const ny = y + dy[i]
            if (nx>=0 && nx<m && ny>=0 && ny<n){
                if (!visited[ny][nx] && maps[ny][nx] === 1){
                    q.push([nx,ny,count+1])
                    visited[ny][nx] = true
                }
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/1844. 게임 맵 최단거리/게임 맵 최단거리.py
from collections import deque

def solution(maps):
    rows = len(maps)
    cols = len(maps[0])
    q = deque([(0,0,1)])
    dx = [1,-1,0,0]
    dy = [0,0,1,-1]
    visited = [[0 for _ in range(cols)] for _ in range(rows)]
    visited[0][0] = 1
    while q:
        x,y,cnt = q.popleft()
        
        if x == cols-1 and y == rows-1:
            return cnt
        
        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]
            if 0 <= nx < cols and 0 <= ny and ny < rows:
                if not visited[ny][nx] and maps[ny][nx]:
                    visited[ny][nx] = 1
                    q.append((nx,ny,cnt+1))
    return -1